pkg/pipeline/templates/terraformed.go.tmpl

```diff
@@ -38,6 +38,16 @@ func (tr *{{ .CRD.Kind }}) GetObservation() (map[string]interface{}, error) {
 
 // SetObservation for this {{ .CRD.Kind }}
 func (tr *{{ .CRD.Kind }}) SetObservation(obs map[string]interface{}) error {
```

Comment:
`GetID() string` could be a pretty useful addition to `resource.Terraformed` interface that will allow us use it in references without marshaling.

Reply:
Actually it'd be a nice-to-have since we can cast to the exact type without marshaling anyway, just like in native providers. Perfectly fine to not add it if you don't want to make a change to that interface.

Reply:
I think it's fine to add considering this interface is not in broad use in the wild. If agreed, I think we should be fine to introduce a backward-incompatible change at this maturity level.